operator/controller/dnsnameresolver/resolver.go

```diff
@@ -35,37 +34,44 @@ type dnsResolvedName struct {
 // requests to CoreDNS pods whenever TTL of any IP address associated to a DNS name expires.
 type Resolver struct {
 	corednsEndpointSliceCache cache.Cache
-	serviceName               string
-	dnsNames                  map[string]dnsResolvedName
+	port                      string
+	dnsNames                  map[string]*dnsResolvedName
 	regularObjInfo            map[string]string
 	wildcardObjInfo           map[string]sets.Set[string]
 	dnsLock                   sync.Mutex
 
-	added   chan struct{}
+	added   chan resolvedAddressDetails
 	deleted chan string
```

Comment:
What do you think about unification of 2 channels? Something like:

```go
added chan dnsDetails
deleted chan dnsDetails
```

Where `dnsDetails` is renamed `resolvedAddressDetails`.

Then when you Add or Delete in controller you can do:

```go
resolver.AddName(dnsDetials{dnsName, resolvedAddresses, matchesRegulr, objName})
resolved.DeleteName(dnsDetails{objName})
```